File: app/src/main/java/com/loskon/noteminimalism3/rv/NoteViewHolder.java
package com.loskon.noteminimalism3.rv;

import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import com.loskon.noteminimalism3.R;

/**
 * Наш собственный ViewHolder, который предоставляет доступ к View-компонентам
 */

public class NoteViewHolder extends RecyclerView.ViewHolder {

    LinearLayout linearLayoutCard;
    CardView cardView;
    TextView title;
    TextView date;
    View view;

    public NoteViewHolder(@NonNull View itemView) {
        super(itemView);

        title = itemView.findViewById(R.id.title_item_note);
        date = itemView.findViewById(R.id.date_item_note);

        linearLayoutCard = itemView.findViewById(R.id.linLayCard);
        cardView = itemView.findViewById(R.id.card_view);
        view = itemView.findViewById(R.id.view);
    }
}

File: app/src/main/java/com/loskon/noteminimalism3/ui/recyclerview/notes/NoteViewHolder.kt
package com.loskon.noteminimalism3.ui.recyclerview.notes

import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.loskon.noteminimalism3.databinding.RowNoteBinding
import com.loskon.noteminimalism3.model.Note

/**
 * Доступ к View-компонентам
 */

class NoteViewHolder(private val binding: RowNoteBinding) :
    RecyclerView.ViewHolder(binding.root) {

    fun bind(note: Note) {
        binding.note = note
        binding.executePendingBindings() // Используется, чтобы биндинг не откладывался
    }

    // Getters
    val linearLayout: LinearLayout
        get() {
            return binding.linearLayoutRowNote
        }

    val viewFavorite: View
        get() {
            return binding.viewFavorite
        }

    val title: TextView
        get() {
            return binding.tvCardNoteTitle
        }

    val date: TextView
        get() {
            return binding.tvCardNoteDate
        }

    val card: CardView
        get() {
            return binding.cardViewNote
        }
}
